create the block of eca in Dev 1.x

wqe123321 opened the pull request:
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

I want to add an attention module to the Neck part, in order to enhance the attention to features.

## Modification

code is following：

import torch.nn as nn
import torch.nn.functional as F
from mmcv.cnn import ConvModule
from mmcv.runner import BaseModule, auto_fp16

# from ..builder import NECKS
from mmrotate.models.builder import ROTATED_NECKS

class eca_layer(nn.Module):#eca注意力
    def __init__(self, channel, k_size=3):
        super(eca_layer, self).__init__()  # super类的作用是继承的时候，调用含super的哥哥的基类__init__函数。
        self.avg_pool = nn.AdaptiveAvgPool2d(1)  # 全局平均池化
        self.max_pool = nn.AdaptiveMaxPool2d(1)
        self.conv = nn.Conv1d(1, 1, kernel_size=k_size,
                              padding=(k_size - 1) // 2, bias=False)  # 基于1*1卷积学习通道之间的信息
        self.sigmoid = nn.Sigmoid()  # 激活函数

    def forward(self, x):
        # x: input features with shape [b, c, h, w]
        b, c, h, w = x.size()  # b代表b个样本，c为通道数，h为高度，w为宽度
        # feature descriptor on the global spatial information
        y = self.avg_pool(x)
        # Two different branches of ECA module
        # torch.squeeze()这个函数主要对数据的维度进行压缩,torch.unsqueeze()这个函数 主要是对数据维度进行扩充
        y = self.conv(y.squeeze(-1).transpose(-1, -2)).transpose(-1, -2).unsqueeze(-1)
        # Multi-scale information fusion多尺度信息融合
        y = self.sigmoid(y)
        # 原网络中克罗内克积，也叫张量积，为两个任意大小矩阵间的运算
        return x * y.expand_as(x)

@ROTATED_NECKS.register_module()
class ECA(BaseModule):   #eca注意力
    def __init__(self,
                 in_channels,
                 out_channels,
                 num_outs,
                 start_level=0,
                 end_level=-1,
                 add_extra_convs=False,
                 relu_before_extra_convs=False,
                 no_norm_on_lateral=False,
                 conv_cfg=None,
                 norm_cfg=None,
                 act_cfg=None,
                 upsample_cfg=dict(mode='nearest'),
                 init_cfg=dict(
                     type='Xavier', layer='Conv2d', distribution='uniform')):
        super(FPN_eca, self).__init__(init_cfg)
        assert isinstance(in_channels, list)
        self.in_channels = in_channels
        self.out_channels = out_channels
        self.num_ins = len(in_channels)
        self.num_outs = num_outs
        self.relu_before_extra_convs = relu_before_extra_convs
        self.no_norm_on_lateral = no_norm_on_lateral
        self.fp16_enabled = False
        self.upsample_cfg = upsample_cfg.copy()

        if end_level == -1 or end_level == self.num_ins - 1:
            self.backbone_end_level = self.num_ins
            assert num_outs >= self.num_ins - start_level
        else:
            # if end_level is not the last level, no extra level is allowed
            self.backbone_end_level = end_level + 1
            assert end_level < self.num_ins
            assert num_outs == end_level - start_level + 1
        self.start_level = start_level
        self.end_level = end_level
        self.add_extra_convs = add_extra_convs
        assert isinstance(add_extra_convs, (str, bool))
        if isinstance(add_extra_convs, str):
            # Extra_convs_source choices: 'on_input', 'on_lateral', 'on_output'
            assert add_extra_convs in ('on_input', 'on_lateral', 'on_output')
        elif add_extra_convs:  # True
            self.add_extra_convs = 'on_input'

        self.lateral_convs = nn.ModuleList()
        self.fpn_convs = nn.ModuleList()
        self.fpn_att = nn.ModuleList()#ModeleList,可以存放很多函数，调用的时候类似数组用下标调用，比如fpn_att[i]
    

        for i in range(self.start_level, self.backbone_end_level):#遍历特征图，为了获取特征图通道数
            l_conv = ConvModule(
                in_channels[i],  # 输入通道数
                out_channels,#输出通道数
                1,
                conv_cfg=conv_cfg,
                norm_cfg=norm_cfg if not self.no_norm_on_lateral else None,
                act_cfg=act_cfg,
                inplace=False)
            fpn_conv = ConvModule(
                out_channels,
                out_channels,
                3,
                padding=1,
                conv_cfg=conv_cfg,
                norm_cfg=norm_cfg,
                act_cfg=act_cfg,
                inplace=False)
            s_layer = eca_layer(out_channels)

            self.lateral_convs.append(l_conv)
            self.fpn_convs.append(fpn_conv)
            self.fpn_att.append(s_layer)#容器fpn_att存放了每一层特征图的注意力函数，因为每层注意力输入通道数不一样
#包含了每一个特征图的操作
        # add extra conv layers (e.g., RetinaNet)
        extra_levels = num_outs - self.backbone_end_level + self.start_level
        if self.add_extra_convs and extra_levels >= 1:
            for i in range(extra_levels):
                if i == 0 and self.add_extra_convs == 'on_input':
                    in_channels = self.in_channels[self.backbone_end_level - 1]
                else:
                    in_channels = out_channels
                extra_fpn_conv = ConvModule(
                    in_channels,
                    out_channels,
                    3,
                    stride=2,
                    padding=1,
                    conv_cfg=conv_cfg,
                    norm_cfg=norm_cfg,
                    act_cfg=act_cfg,
                    inplace=False)
                self.fpn_convs.append(extra_fpn_conv)

    @auto_fp16()
    def forward(self, inputs):
        """Forward function."""
        assert len(inputs) == len(self.in_channels)

        # build laterals
        laterals = [#backbone输出特征图
            lateral_conv(inputs[i + self.start_level])
            for i, lateral_conv in enumerate(self.lateral_convs)
        ]
        laterals[0] = self.fpn_att[0](laterals[0])#将原始特征图经过注意力操作后，覆盖原始特征图，得到新的laterals[0]
        laterals[1] = self.fpn_att[1](laterals[1])
        laterals[2] = self.fpn_att[2](laterals[2])
        laterals[3] = self.fpn_att[3](laterals[3])
            #这里才是对特征图进行操作！！！！！！！！！！！！！！！！！！！！
        # build top-down path
        used_backbone_levels = len(laterals)
        for i in range(used_backbone_levels - 1, 0, -1):
            # In some cases, fixing `scale factor` (e.g. 2) is preferred, but
            #  it cannot co-exist with `size` in `F.interpolate`.
            if 'scale_factor' in self.upsample_cfg:
                # fix runtime error of "+=" inplace operation in PyTorch 1.10
                laterals[i - 1] = laterals[i - 1] + F.interpolate(
                    laterals[i], **self.upsample_cfg)
            else:
                prev_shape = laterals[i - 1].shape[2:]
                laterals[i - 1] = laterals[i - 1] + F.interpolate(
                    laterals[i], size=prev_shape, **self.upsample_cfg)

        # build outputs
        # part 1: from original levels
        outs = [
            self.fpn_convs[i](laterals[i]) for i in range(used_backbone_levels)
        ]
        # part 2: add extra levels
        if self.num_outs > len(outs):
            # use max pool to get more levels on top of outputs
            # (e.g., Faster R-CNN, Mask R-CNN)
            if not self.add_extra_convs:
                for i in range(self.num_outs - used_backbone_levels):
                    outs.append(F.max_pool2d(outs[-1], 1, stride=2))
            # add conv layers on top of original feature maps (RetinaNet)
            else:
                if self.add_extra_convs == 'on_input':
                    extra_source = inputs[self.backbone_end_level - 1]
                elif self.add_extra_convs == 'on_lateral':
                    extra_source = laterals[-1]
                elif self.add_extra_convs == 'on_output':
                    extra_source = outs[-1]
                else:
                    raise NotImplementedError
                outs.append(self.fpn_convs[used_backbone_levels](extra_source))
                for i in range(used_backbone_levels + 1, self.num_outs):
                    if self.relu_before_extra_convs:
                        outs.append(self.fpn_convs[i](F.relu(outs[-1])))
                    else:
                        outs.append(self.fpn_convs[i](outs[-1]))
        return tuple(outs)

zytx121 committed:
[Refactor] add circle ci (#438)

* init

* delete github action

* test

* Update version.py

* Update version.py

zytx121 committed:
[Refactor] clean UT (#441)

* delete UT

* revert

* update

* Update test.yml

zytx121 committed:
[Refactor] refactor DOTA Dataset&Metric (#446)

* update

* fix lint

* update ut

* fix

* Update dota.py

* Update dota.py

* Update anchor_generator.py

* Update dota.py

* Update registry.py

* update metric

* add format ut

* add merge_patches in DOTAMetric

* Update dota_metric.py

* Update mmrotate/datasets/dota.py

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

* Update mmrotate/evaluation/metrics/dota_metric.py

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

* Update dota_metric.py

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

zytx121 committed:
[Refactor] default runtime config (#447)

* init

* Update default_runtime.py

jbwang1997 committed:
[Feature] Add RotatedBoxes and QuadriBoxes data structure in mmrotate. (#450)

* Initialize RotatedBoxes and QuadriBoxes

* Move all box modes into Structures, Add docstring

* Update docstring

* Upload UT

* Fix comments

* Update inplace operations

* Update

* Update

* Update

* Update

* Updata rotate and project UT

* Update

* Update

* Update

* Update

* update

* Update from_instance_masks

zytx121 committed:
[Refactor] refactor HRSCDataset (#457)

* add

* add

* Update dota_metric.py

* update

* fix lint

* remove hrsc metric

* fix typo

* Update dota_metric.py

* Update mmrotate/datasets/hrsc.py

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

* Update mmrotate/datasets/hrsc.py

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

* Update hrsc.py

* Update hrsc.py

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

zytx121 committed:
[Refactor] refactor RetinaNet based on boxlist (#472)

* init

* fix lint

zytx121 committed:
[Refactor] refactor faster rcnn (#508)

* init

* fix lint

* update

* add  UT

* fix lint

* update coder

* Update mmrotate/core/bbox/iou_calculators/rotate_iou2d_calculator.py

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

* fix docstring

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

zytx121 committed:
[Refactor] refactor gwd&kld (#509)

* init

* fix lint

* update

* update coder

* init

* fix lint

* fix bug in DOTA metric

* Update delta_xywht_hbbox_coder.py

* support tensor input when coder.decode.

In two stage detectors and refine single stage detectors, the bboxes can be Tensor.

* Update rotated_retinanet_obb_r50_fpn_1x_dota_le90.py

* support use_box_type in coder

* fix bug

* fix

zytx121 committed:
[Refactor] R3Det and S2A-Net (#514)

* init

* fix lint

* update

* update coder

* init

* update

* update ut

* fix lint

* Update delta_xywht_hbbox_coder.py

* Update delta_xywht_hbbox_coder.py

* update ut

* fix docstring

* fix

* update

* fix

* add UT

* Create cascade_s2anet_r50_fpn_1x_dota_le135.py

* Update refine_single_stage.py

* Update refine_single_stage.py

* fix bug when run demo image

* Update refine_single_stage.py

* fix bug

* Update refine_single_stage.py

* fix docstring

zytx121 committed:
Refactor oriented rcnn (#515)

* update

* Update test_two_stage.py

* Update s2anet_r50_fpn_fp16_1x_dota_le135.py

* add UT

* Update test_delta_midpointoffset_rbbox_coder.py

* update

* Update oriented_rcnn_r50_fpn_fp16_1x_dota_le90.py

zytx121 committed:
Refactor roi trans & redet (#516)

* Refactor oriented rcnn (#515)

* update

* Update test_two_stage.py

* Update s2anet_r50_fpn_fp16_1x_dota_le135.py

* add UT

* Update test_delta_midpointoffset_rbbox_coder.py

* update

* Update oriented_rcnn_r50_fpn_fp16_1x_dota_le90.py

* update

* update

* fix lint

* Update re_resnet.py

* Update re_resnet.py

zytx121 committed:
[Refactor] refactor Rotated RepPoints (#523)

* init

* add UT

* fix lint

* Update rotated_reppoints_head.py

* Update rotated_reppoints_r50_fpn_1x_dota_qbox.py

* fix

zytx121 committed:
[Refactor] CFA & SASM (#524)

* init

* fix bug

* add UT

* fix lint

* fix docstring

* update

* fix

zytx121 committed:
[Refactor] oriented reppoints (#530)

* add UT

* fix lint

* init

* fix

* Update oriented_reppoints_head.py

* update

* Update test_oriented_reppoints.py

* fix error when train

* fix UT

* update docstring

* Update mmrotate/models/dense_heads/oriented_reppoints_head.py

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

* Update mmrotate/models/dense_heads/oriented_reppoints_head.py

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

* Update mmrotate/models/dense_heads/oriented_reppoints_head.py

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

* Update oriented_reppoints_head.py

* upgrade pre-commit-hook

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

zytx121 committed:
[Refactor] CSL (#535)

* add UT

* fix lint

* update

* Update test_oriented_reppoints.py

* fix UT

* init

* Update angle_branch_retina_head.py

* add ut

* Update test_angle_coder.py

* add UT

* Update test_angle_branch_retina_head.py

* fix

* fix

k-papadakis committed:
Fix: collections import error for python3.10 (#533)

ABCs were moved from collections to collections.abc in 3.3.
Since 3.10, ABCs cannot be imported directly from collections.

k-papadakis committed:
confusion_matrix.py bugfixes (#531)

1. Removed invalid `score_threshold` arg  in `nms_rotated` call
2. Passed `--color-theme` to `plot_confusion_matrix`

liuyanyi committed:
[Refactor] Rotated FCOS (#543)

* fcos for 1.x

* update config

* add test

* fix config

* fix

* fix

* fix

* remove useless class

zytx121 committed:
[Refactor] kfiou (#536)

* add UT

* fix lint

* update

* Update test_oriented_reppoints.py

* fix UT

* init

* fix

* add UT

* fix lint

* update

* fix

liuyanyi committed:
fix channel (#545)

zytx121 committed:
[Refactor] Gliding vertex (qbox version) (#520)

* Update s2anet_r50_fpn_fp16_1x_dota_le135.py

* add UT

* Update test_delta_midpointoffset_rbbox_coder.py

* update

* fix lint

* update

* update

* add rbox version

* fix

* add UT

* fix lint

zytx121 committed:
Update __init__.py (#550)

zytx121 committed:
[Refactor] refactor Rotated ATSS (#551)

* old version

* delete fake atss

* add ut

* Update UT

* update docstring

zytx121 committed:
Refactor docs (#552)

* delete 0.x docs

* init new en/docs

* fix lint

* fix docs error

* Update changelog_v0.x.md

* update get_started.py

* Update get_started.md

* Update docs/en/conf.py

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

* update en

* update zh_CN

* Update conf.py

* fix

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

zytx121 committed:
Refactor structure of mmrotate (#558)

* update

* fix lint

* update

* update

* fix ut

* Update test_angle_coder.py

* fix

* Update __init__.py

* fix

* fix ut bugs

zytx121 committed:
[Refactor] rename configs (#560)

* update

* update

* fix lint

* update

* update

* Update gliding-vertex-rbox_r50_fpn_1x_dota.py

* update ms configs

* update hrsc configs

* fix lint

* Update configs/redet/README.md

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

liuyanyi committed:
[Feature] Add RotatedCocoMetric (#557)

* Add Rotated COCO Metric

* Fix

Co-authored-by: Yue Zhou <592267829@qq.com>

zytx121 committed:
Update issue templates (#569)

* update issue template

* Update .github/ISSUE_TEMPLATE/1-bug-report.yml

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

zytx121 committed:
Refactor READEME (#566)

* update

* update citation

* update

* update

* update

* update

DonggeunYu committed:
[FIX] Import RotatedBoxes from structures not core (#572)

[FIX] Import RotatedBoxes from structures not core

DonggeunYu committed:
[FIX] Fix a typo in the config file. (#571)

zytx121 committed:
Fix kfiou loss nan bug. (#555)

* fix bug

* Update kf_iou_loss.py

jbwang1997 committed:
[Refactor] Refactor image_demo and huge_image_demo (#561)

* init

* Fix comments

* Fix comments

* Update

* mv patch into mmrotate.utils

* Update

* Update

zytx121 committed:
Refactor visualizer (#581)

* update

* fix lint

* update

zytx121 committed:
Support Coco style annotations (#582)

* update

* Update transforms.py

* update

* Update dota_coco.py

* update

* Update README.md

* update

* Update dota_coco.py

* Update dota_coco.py

zytx121 committed:
Refactor tools (#590)

* update tools

* fix lint

* Update confusion_matrix.py

* update

zytx121 committed:
Refactor sar dataset (#591)

* init

* update

* fix lint

* Update hrsid

* update link

* update

zytx121 committed:
Update CI for 1.x & Improve the codecov to 80% (#596)

* update CI

* skip some ci when no cuda

* fix ci

* fix CircleCI

* fix ci error

* fix CircleCI

* fix

* Update angle_coder.py

* fix ci in windows

* fix windows ci

* update

* update

* Update angle_coder.py

* update

* update codecov

* Update config.yml

* update

* Update test.yml

* Update test.yml

* update

* Update test.yml

* Update test.yml

* Update test.yml

* Update test.yml

* delete useless things

* improve codecov from 72% -> 78%

* update

* Add UT

* fix lint

* update

* Update transforms.py

zytx121 committed:
Changelog v1.0.0rc0 (#599)

* skip some ci when no cuda

* fix ci in windows

* fix windows ci

* update

* update codecov

* Update config.yml

* update

* Update test.yml

* Update test.yml

* update

* Update test.yml

* Update test.yml

* Update test.yml

* Update test.yml

* improve codecov from 72% -> 78%

* update

* fix lint

* Docs. (To be updated)

* update

* Update

* update

* Update README.md

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

* Update README.md

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

* fix

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

RangiLyu committed:
[Doc] Update README.md for 1.x. (#602)

zytx121 committed:
Update readme of 1.0.0rc0 (#603)

* update

* Update README_zh-CN.md

zytx121 committed:
Fix image_demo bug. (#613)

* update

* Update local_visualizer.py

austinmw committed:
use mmengine in torchserve deployment (#616)

* use mmengine

* add precommit fix

zytx121 committed:
Fix rbbox_overlaps error (#620)

* Update bbox_overlaps.py

* add UT

* Update bbox_overlaps.py

RangiLyu committed:
fix windows ci (#621)

* fix windows ci

* fix windows ci

zytx121 committed:
[Community] Add 'Projects/' folder, and the first example project (#627)

* init

* Update r3det-oc_dummy-resnet_fpn_1x_dota.py

* Update README.md

zytx121 committed:
Update test.py (#636)

RangeKing committed:
Create .pre-commit-config-zh-cn.yaml (#630)

YanxingLiu committed:
add mmrotate 1.x dockerfile (#631)

* add mmrotate 1.x dockerfile

* Update docker/Dockerfile

Co-authored-by: Range King <RangeKingHZ@gmail.com>

* Update docker/Dockerfile

Co-authored-by: Range King <RangeKingHZ@gmail.com>

* Update docker/Dockerfile

use 1.x main branch instead of dev branch

Co-authored-by: Hakjin Lee <nijkah@gmail.com>

Co-authored-by: Range King <RangeKingHZ@gmail.com>
Co-authored-by: Hakjin Lee <nijkah@gmail.com>

yuyi1005 committed:
[Algorithm] Support PSC (#617)

* [Algorithm] Support PSC

Support PSC, phase-shifting coder and dual-frequency phase-shifting coder.

* Change encoded_size to encode_size

* Add configs based on rotated-fcos

* Apply suggestions from code review

Co-authored-by: Yue Zhou <592267829@qq.com>

* Add HRSC config for rotated_fcos

* Update README, Fix bug in AngleBranchRetinaHead

* Fix README lint error

* Update README

* Update download link

* Add metafile

* Update model-index.yml

* Update metafile.yml

Co-authored-by: Yue Zhou <592267829@qq.com>

zytx121 committed:
Update model-index.yml (#637)

yxzhao2022 committed:
[Fix] fix get_flops.py (#646)

yxzhao2022 committed:
[Feature] Support DIOR Dataset (#639)

* [Feature] Support DIOR Dataset

* [Feature] Support DIOR Dataset

* [Feature] Support DIOR Dataset

* [Feature] Support DIOR Dataset

* [Feature] Support DIOR Dataset

* [Feature] Support DIOR Dataset

* [Feature] Support DIOR Dataset

* [Feature] Support DIOR Dataset

* Update test_dior.py

* Update dior.py

* Update README.md

* Update metafile.yml

yangxue0827 committed:
[Algorithm] Support H2RBox (#644)

* [Algorithm] Support H2RBox

* upload config and readme

* Update dota.py

* update unit loss test

* fix ci

* delete useless codes

* update other implementation

* Create test_h2rbox_head.py

* Upload 3x config

* Update 3x performance

* support dotav1.5 and dotav2

* Update h2rbox performance on dotav15

* Update dotav2

* Update dotav2.py

* Update ms performance on dota

* Update dotav2.py

* Create h2rbox-le90_r50_fpn_adamw-1x_dotav2.py

* Update performance on dotav2

* Create h2rbox-le90_r50_fpn_adamw-3x_dotav15.py

* Update 3x performance on dotav15

* Create h2rbox-le90_r50_fpn_adamw-3x_dotav2.py

* update 3x performance on dotav2

* Add H2Rbox in README

* Add h2rbox in model zoo

* Update test_dota.py

* Create metafile.yml

* Update weights and logs

* Fix

* Update __init__.py

* Create h2rbox-le90_r50_fpn_adamw-1x_dior.py

* Update performance on dior

* Fix wrong param name

Co-authored-by: Yue Zhou <592267829@qq.com>

nijkah committed:
[Fix][dev-1.x] sklearn depreated, install scikit-learn (#659)

* [Fix] sklearn depreated, install scikit-learn

* lint

* fix sklearn

liuyanyi committed:
CodeCamp #38 RTMDet-R (#662)

* Add DistributionAngleCoder

* fix keepdim

* rtm init

* fix distribution angle coder

* clean

* clean

* add probiou

* add hrsc cfg

* fix GDLoss with empty input

* add probiou cfg

* add mixup mosaic cfg

* Add DistributionAngleCoder

* fix keepdim

* rtm init

* fix distribution angle coder

* clean

* clean

* add probiou

* add hrsc cfg

* fix GDLoss with empty input

* add probiou cfg

* add mixup mosaic cfg

* fix mmdeploy vis error

* add function_rewriter for deploy

* fix hrsc config

* remove deploy

* move distance2obb to transforms, remove useless code

* refactor configs

* fix readme

* fix readme

* clean

* add docstring

* add unittest

* update metafile

* remove useless code

* add img height

* typo

* update results

* fix table

* optimize tables

* add

* fix config

* upload

* update pth link

* add readme pth

* update readme

* update readme

* remove useless cfg

* fix ut on cpu

* fix comment

* add comment for change

* add cite

zytx121 committed:
[CI] Add torch 1.13 (#661)

zytx121 committed:
Modify keys in dataset.metainfo to lower case. (#654)

* init

* Update dota.py

RangeKing committed:
[Fix] Deprecating old type alias due to new version of numpy (#674)

* deprecating `np.int` type alias

* deprecating `np.bool` type alias

JosonChan1998 committed:
[Enhance]: add data gitignore (#668)

zytx121 committed:
Bump version to 1.0.0rc1 (#676)

* update

* fix lint

* update

* update

* update

* Update test.yml

* Update __init__.py

* fix CLASSES

RangiLyu committed:
limit numpy version (#682)

zytx121 committed:
Fix CI error when 'numpy >= 1.24.0' (#683)

* Update merge_stage_test.yml

* Update re_fpn.py

* Update re_fpn.py

liuyanyi committed:
[Fix] Add missing models in model index (#702)

* fix model-index

* add missing model index

* fix convnext

* fix h2rbox

qianlian-mozi committed:
Update get_started.md (#694)

完成了文档的翻译，但是请注意我没有搞懂 []() 的链接方式，翻译后的文档中在“需要准备的依赖环境”下的note中有一个失败的链接；在“在Google Colab上安装”的第三步中也有一个失败的链接与一次失败的``引用；我很苦恼解决不了这三个问题。

zytx121 committed:
Update ICLR2023 papers (#718)

* update

* date

* year

* update

liuyanyi committed:
fix img_shape (#719)

zytx121 committed:
Set `e2cnn` as an optional dependency (#724)

* update

* update

* update

* update

* update

* update

* Update re_resnet.py

* fix lint

* Update test.yml

* fix circleCI

* Update test.yml

* Update test.yml

vansin committed:
docs: Add twitter discord medium youtube link (#729)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/open-mmlab/mmrotate?pullRequest=737) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/open-mmlab/mmrotate?pullRequest=737) before we can accept your contribution.<br/>**9** out of **16** committers have signed the CLA.<br/><br/>:white_check_mark: zytx121<br/>:white_check_mark: RangiLyu<br/>:white_check_mark: liuyanyi<br/>:white_check_mark: yxzhao2022<br/>:white_check_mark: YanxingLiu<br/>:white_check_mark: JosonChan1998<br/>:white_check_mark: qianlian-mozi<br/>:white_check_mark: vansin<br/>:white_check_mark: nijkah<br/>:x: jbwang1997<br/>:x: k-papadakis<br/>:x: DonggeunYu<br/>:x: austinmw<br/>:x: yuyi1005<br/>:x: RangeKing<br/>:x: yangxue0827<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/open-mmlab/mmrotate?pullRequest=737) it.</sub>

zytx121 commented:
Hi @xxx,
Thanks for your kind PR. It seems that your branch is out of date and causes much difference.
Now, we require all the PRs to be merged into the development branch, i.e., dev branch for MMRotate 0.x and dev-1.x branch for MMRotate 1.x. 
If your modification is not checkout from these two branches, they may cause conflicts and you should add your modification based on these two branches.
If your modification is checkout from these two branches, your branch might be out of date. In such a case
Please rebase your branch to the target dev or dev-1.x branch. You can simply do that by modifying your pull behavior:
```
git config --local --add pull.rebase true  # make rebase a default behavior in pull
```
Then you can git pull and rebase the dev branch by
```
git pull dev-1.x  # if your modification is for MMRotate 1.x
git pull dev  # if your modification is for MMRotate 0.x
```
After rebase, you might need to add `--force` option when pushing code, e.g.,
```
git push [remote name] [you branch] --force
```

zytx121 committed:
Fix type hint import errors (#742)

zytx121 committed:
Update DOTA dataset for mmyolo (#743)

zytx121 committed:
Add PSC in README.md (#732)

zytx121 committed:
Update CVPR2023 paper (#747)

* update

* Update README.md

vansin committed:
[Projects] Support RR360 (Rotated Rectangle 360) Detection in TRR360D (#731)

* init headet

* feat: add train and test

* feat:update readme

* feat: add h180

* docs: update readme

* feat: update readme

* feat: add datatsets

* docs: update h180 comments

* feat: add configs-ic19-obb-obb configs

* feat: update metric

* feat: update metric

* feat: delete unnecessary

* feat: init rtmdet

* feat: update datasets

* feat: update local_visualizer comments

* feat: add benchmark.py

* docs: update

* docs: update

* feat: update angle h180 in rtmdet

* feat: add roi_trans

* feat: add 180

* docs: update RotatedBoxes

* feat: add r_l1_loss

* feat: add rl1 loss config

* feat: update checkpoint hook

* feat: change the weight

* feat: update the loss weight 5

* fix: reletive import

* feat: fix the APH metric

* fix: fix the regularize_boxes

* fix: fix the edge_swap=False

* feat: init the tests in headet

* feat: strict the loss

* update interval

* feat: update logger config

* feat: support rtmdet 360

* feat: delete unnecessery comment

* feat: init the ic19 datasets

* feat: init ic19 datasets

* feat: add rotated_rtmdet_x3

* feat: add rotated_rtmdet_x3_r

* feat: update rtmdet

* feat: add a1l1 loss

* docs: update readme

* rename headet to RR360 Det

* feat: refactor the config

* docs: update readme

* fix: headet to RR360

* fix: delete unuse

* add test script

* feat: update readme

* fdocs: update script

* docs: update

* docs: update

* docs: update

* docs: update

* docs: update

* remove the configs180

* docs: update

* delete the unuse

* delete the unuse

* delete the unuse

* feat: delete the unuse

* docs: delete the unuse

* delete unuse

* delete the unuse

* delete the unuse

* fix: replace h180 with r360

* feat: delete the Unnecessary Comments

* feat: add rotated_retinanet

* fix: DOTAHeadMetric to DOTAR360Metric

* fix: hangle_thr to angle_thr

* fix: ORLocalVisualizer to RR360LocalVisualizer

* fix: delete the mmcls.models

* fix: delete the benckmark

* fix: pre-commit

* fix: delete unuse

* fix: dota_head_metric to dota_r360_metric

* fix: AP(T<xxxx)

* [Fix]: mAPH to mAP

* docs: update readme

* docs: update

* fix: ICDAR2019_MTD_HOQ to TRR360D

* fix: delete the unuse a l1 loss

* fix: delete the unuse model import

* docs: update

* fix: disable wandb

* feat: add without l1_loss

* docs: update en docs

* docs: update the datasets link

* docs: add todo

zytx121 committed:
Fix loss NaN bug when training HRSID dataset. (#760)

* Update hrsid.py

* Update configs/_base_/datasets/hrsid.py

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

---------

Co-authored-by: RangiLyu <lyuchqi@gmail.com>

DapengFeng committed:
[Fix] Move points_range to the same device with points (#784)

zytx121 committed:
Fix bug when training images without gt (#786)

zcablii committed:
[Project] Support LSKNet (#782)

* LSKNet implementation

* passed by pre-commit

* update configs

* Update lsknet.py

* Update projects/LSKNet/README.md

Co-authored-by: Yue Zhou <592267829@qq.com>

* Update projects/LSKNet/README.md

Co-authored-by: Yue Zhou <592267829@qq.com>

* Update projects/LSKNet/README.md

Co-authored-by: Yue Zhou <592267829@qq.com>

* Update projects/LSKNet/README.md

Co-authored-by: Yue Zhou <592267829@qq.com>

* Update projects/LSKNet/README.md

Co-authored-by: Yue Zhou <592267829@qq.com>

* update

* update ckpt download link

* update ckpt links

* update pretrained weight link

* update pretrained weight link

* update pretrained weight link

* update pretrained weight link

* pre-commit checked

---------

Co-authored-by: Yue Zhou <592267829@qq.com>

zytx121 committed:
[Feature] File I/O migration and reconstruction (#783)

* update

* Update dota_coco.py

* update

* fix

fengshiwest committed:
[doc] add social network links (#789)

* [doc]:add more social network links

* [doc]:add more social network links

OpenMMLab-Assistant001 commented:
Hi @wqe123321  ！We are grateful for your efforts in helping improve this open-source project during your personal time.

Welcome to join OpenMMLab Special Interest Group (SIG) private channel on **Discord**, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add **"mmsig + Github ID"** as a remark when adding friends：）

Thank you again for your contribution❤

yuyi1005 committed:
Support H2RBox-v2 (#805)

* Support H2RBox-v2

* Update README

* Add figure in README

* Update figure in README

* Fix README lint problem

fengshiwest committed:
[Feature] Semi automated with label studio (#815)

* [doc]:add more social network links

* [doc]:add more social network links

* [Feature] support semi-automatic annotation base Label-Studio

* [fix] change the name and url to mmrotate

* [fix] resolve conflict

kitecats committed:
update readme of playground project (#834)

liuyanyi committed:
[Fix] fix LSKNet error in import (#839)

* fix lsknet

* fix some comment

jamiechoi1995 committed:
[Fix] Update train.py to support distributed training with PyTorch>= 2.0.0 (#860)

Li-Qingyun committed:
update README.md (#855)

liuyanyi committed:
[Fix] Fix confusion_matrix (#870)

* fix confusion_matrix

* fix opencv error

Li-Qingyun committed:
fix box error in h2rbox_head.py (#881)

There seems to be a lack of indentation, resulting in an assignment bug when square classes are not processed.

jamiechoi1995 committed:
[Fix] Fix nan loss in RotatedIoULoss when using AmpOptimizer (#889)

* [Fix] Fix nan loss in RotatedIoULoss when using AmpOptimizer

* Fix lint

CSberlin committed:
refine tutorial (#899)

Li-Qingyun committed:
[Fix] update mmdet maximum version to enable latest mmdet-3.1.0 (#918)

* update version in __init__.py

* update version in mminstall.txt

* update version in en faq.md

* update version in zh_cn faq.md

crazysteeaam committed:
Update README_zh-CN.md (#923)

Li-Qingyun committed:
Update README.md of h2rboxv2

Li-Qingyun committed:
Update homepage

Li-Qingyun committed:
Update abs

Li-Qingyun committed:
Update abs

yuyi1005 committed:
Update README.md

yuyi1005 committed:
Merge pull request #952 from Li-Qingyun/patch-2

Update NIPS information of h2rbox-v2
